probe-rs/src/semihosting.rs

```diff
@@ -72,29 +71,48 @@ pub struct GetCommandLineRequest(Buffer);
 impl GetCommandLineRequest {
     /// Writes the command line to the target. You have to continue the core manually afterwards.
     pub fn write_command_line_to_target(
-        &mut self,
+        &self,
         core: &mut dyn CoreInterface,
         cmdline: &str,
     ) -> Result<()> {
         let mut buf = cmdline.to_owned().into_bytes();
         buf.push(0);
-        self.0.write(core, &buf)
+        self.0.write(core, &buf)?;
+
+        // signal to target: status = success
+        write_semihosting_return_value(core, 0)?;
+
+        Ok(())
+    }
+}
+
+fn write_semihosting_return_value(core: &mut dyn CoreInterface, value: u32) -> Result<()> {
+    // different register depending on architecture (and semihosting spec)
+    match core.architecture() {
+        probe_rs_target::Architecture::Xtensa => {
```

Comment:
https://github.com/probe-rs/probe-rs/pull/2303#discussion_r1532575383 We should fix this to not use `RegisterId` before merging.

Reply:
I'm fine with doing this separately, I just don't want us to become used to this particular smell :)

Reply:
It was a hack we left in, so we should fix it :D. @SergioGasquez would you mind trying to get this working again? It'll be worth printing what probe-rs thinks argument register 2 is, it's possible there is some mismatch somewhere.